"Share Feedback" modal triggers the "Discard feedback?" modal when escape is pressed in Vim mode

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The **Share Feedback** modal triggers the **Discard feedback?** modal when escape is pressed in Vim mode.

#### Reproduction

1. Configure Zed to use Vim mode.
2. Click the **Share Feedback** button in the dock.
3. Click into the text area.
4. Hit **i** to enter insert mode.
5. Type some text.
6. Hit **Escape** to exit insert mode.
7. Observe the **Discard feedback?** modal pops up.

#### Expected Behavior

The **Discard feedback?** modal should not pop up unless:
- The user is already in command mode in Vim.
- The user clicks the **Cancel** button.

### Environment

Zed: v0.119.20 (Zed)
OS: macOS 14.3.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="1496" alt="Screenshot 2024-01-29 at 13 11 50" src="https://github.com/zed-industries/zed/assets/24284972/4b38c7dd-770b-4a25-a50a-994b3b9c7ca8">


### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.


```
2024-01-29T18:26:59 [INFO] ========== starting zed ==========
2024-01-29T18:26:59 [INFO] Opening main db
2024-01-29T18:26:59 [INFO] set environment variables from shell:/opt/homebrew/bin/fish, path:REDACTED
2024-01-29T18:26:59 [ERROR] crates/zed/src/zed.rs:538: EOF while parsing a value at line 1 column 0
2024-01-29T18:26:59 [INFO] Opening main db
2024-01-29T18:26:59 [INFO] Node runtime install_if_needed
2024-01-29T18:27:00 [INFO] 0 unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-01-29T18:27:00.428Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-01-29T18:27:00.428Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-01-29T18:27:00 [INFO] 0 unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "d676028c-d18a-4336-a8bc-fe4385acced3",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-01-29T18:27:00 [INFO] 0 unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-01-29T18:27:00.436Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-01-29T18:27:00.436Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
```

--- a/crates/editor/src/editor.rs
+++ b/crates/editor/src/editor.rs
@@ -399,6 +399,7 @@ pub struct Editor {
     workspace: Option<(WeakView<Workspace>, i64)>,
     keymap_context_layers: BTreeMap<TypeId, KeyContext>,
     input_enabled: bool,
+    use_modal_editing: bool,
     read_only: bool,
     leader_peer_id: Option<PeerId>,
     remote_id: Option<ViewId>,
@@ -1482,6 +1483,7 @@ impl Editor {
             workspace: None,
             keymap_context_layers: Default::default(),
             input_enabled: true,
+            use_modal_editing: mode == EditorMode::Full,
             read_only: false,
             use_autoclose: true,
             leader_peer_id: None,
@@ -1780,6 +1782,14 @@ impl Editor {
 
     pub fn set_show_copilot_suggestions(&mut self, show_copilot_suggestions: bool) {
         self.show_copilot_suggestions = show_copilot_suggestions;
+    }
+
+    pub fn set_use_modal_editing(&mut self, to: bool) {
+        self.use_modal_editing = to;
+    }
+
+    pub fn use_modal_editing(&self) -> bool {
+        self.use_modal_editing
     }
 
     fn selections_did_change(

--- a/crates/feedback/src/feedback_modal.rs
+++ b/crates/feedback/src/feedback_modal.rs
@@ -187,6 +187,7 @@ impl FeedbackModal {
             editor.set_show_gutter(false, cx);
             editor.set_show_copilot_suggestions(false);
             editor.set_vertical_scroll_margin(5, cx);
+            editor.set_use_modal_editing(false);
             editor
         });
 

--- a/crates/vim/src/vim.rs
+++ b/crates/vim/src/vim.rs
@@ -198,7 +198,7 @@ impl Vim {
     }
 
     fn activate_editor(&mut self, editor: View<Editor>, cx: &mut WindowContext) {
-        if editor.read(cx).mode() != EditorMode::Full {
+        if !editor.read(cx).use_modal_editing() {
             return;
         }
 

